egui/src/widgets/plotting/plot.rs

```diff
@@ -419,18 +171,59 @@ impl Plot {
         self.show_y = show_y;
         self
     }
+
+    /// Always keep the x-axis centered.
+    pub fn center_x_axis(mut self, on: bool) -> Self {
+        self.center_x_axis = on;
+        self
+    }
+
+    /// Always keep the y-axis centered.
+    pub fn center_y_axis(mut self, on: bool) -> Self {
+        self.center_y_axis = on;
+        self
+    }
+
+    /// Whether to allow zooming in the plot.
+    pub fn allow_zoom(mut self, on: bool) -> Self {
+        self.allow_zoom = on;
+        self
+    }
+
+    /// Whether to allow dragging in the plot to move the bounds.
+    pub fn allow_drag(mut self, on: bool) -> Self {
+        self.allow_drag = on;
+        self
+    }
+
+    /// Expand bounds to include the given x value.
+    /// For instance, to always show the y axis, call `plot.include_x(0.0)`.
+    pub fn include_x(mut self, x: impl Into<f64>) -> Self {
+        self.min_auto_bounds.extend_with_x(x.into());
+        self
+    }
+
+    /// Expand bounds to include the given y value.
+    /// For instance, to always show the x axis, call `plot.include_y(0.0)`.
+    pub fn include_y(mut self, y: impl Into<f64>) -> Self {
+        self.min_auto_bounds.extend_with_y(y.into());
+        self
+    }
```

Comment:
the diff becomes harder to read when moving around code like this. what is the motivation?